docs/core/middleware.md

```diff
@@ -0,0 +1,155 @@
+<!--
+order: 3
+-->
+
+# Middlewares
+
+
+`Middlewares` are objects that the developer can create to add logic before or after the transaction handler in a composable manner.
+{synopsis}
+
+## Pre-requisite Readings
+
+* [Anatomy of a Cosmos SDK Application](../basics/app-anatomy.md) {prereq}
+* [Transactons](transactions.md){prereq}
+
+
+## Middlewares
+
+The SDK Baseapp's implementation of ABCI CheckTx, DeliverTx, and Simulate methods use a middleware-based design. Middlewares can add logic to be executed before or after a transaction handler execution. Middlewares allow us to solve use cases like transaction Tips and refund unused gas. 
+
+
+### Type Definition
+
+The two following interfaces are the base of the middleware design, and are defined in types/tx:
+
++++ https://github.com/cosmos/cosmos-sdk/blob/master/types/tx/middleware.go#L62-L71
```

Comment:
```suggestion
+++ https://github.com/cosmos/cosmos-sdk/blob/master/types/tx/middleware.go#L62-L71
```

let's use a commit has instead of master, so that the docs always point to fixed content

Reply:
yes indeed i forgot to ask about this thanks!!!

Reply:
fixed